src/index.ts

```diff
@@ -335,269 +611,46 @@ export function offlineInit(config: IClientConfigSync): EppoClient {
     }
   }
 
-  EppoJSClient.initialized = true;
-  return EppoJSClient.instance;
+  instance.initialized = true;
+  return instance;
 }
 
 /**
- * Tracks pending initialization. After an initialization completes, the value is removed from the map.
+ * Tracks pending initialization. After an initialization completes, this value is nulled
  */
-let initializationPromise: Promise<EppoClient> | null = null;
+let initializationPromise: Promise<EppoJSClient> | null = null;
 
 /**
  * Initializes the Eppo client with configuration parameters.
  * This method should be called once on application startup.
  * If an initialization is in process, calling `init` will return the in-progress
  * `Promise<EppoClient>`. Once the initialization completes, calling `init` again will kick off the
- * initialization routine (if `forceReinitialization` is `true`).
+ * initialization routine (if `forceReinitialize` is `true`).
+ *
+ *
+ * @deprecated
+ * Use `EppoJSClient.createAndInit` instead of `init` and `getInstance`. These will be removed in v4
+ *
  * @param config - client configuration
  * @public
  */
 export async function init(config: IClientConfig): Promise<EppoClient> {
-  validation.validateNotBlank(config.apiKey, 'API key required');
-
-  // If there is already an init in progress for this apiKey, return that.
-  if (!initializationPromise) {
-    initializationPromise = explicitInit(config);
+  if (initializationPromise === null) {
+    initializationPromise = getInstance().init(config);
```

Comment:
minor: again, unrelated to the PR per se but I find this `initializationPromise` disturbing and a potential source of errors.

If `init()` is called with different configuration and `forceReinitialize: true`, the second call is going to be silently discarded. Which may lead to all sorts of troubles as client is now running with unexpected configuration.

With `forceReinitialize: false`, this handling will also silence a warning, so users have less opportunity to catch their bug.

At the very least, the case of concurrent initialization is worth a warning. But overall, I think it might be a good idea to handle `forceReinitialize` here in this function (as per my other comment)

Reply:
This "buffered" initialization was implemented because of massive amounts of traffic from one client. It is one of several measures taken to prevent the flood of requests from improper initialization of the eppo client (ex: calling init everywhere instead of `getInstance`).

I've added a warning here whenever `init` is called but the singleton instance is already initialized (or in progress).